assets/js/app.js

```diff
@@ -77,61 +77,80 @@ document.addEventListener("DOMContentLoaded", () => {
     }
 
     try {
-      while(!doneDisplaying) {
-          page++;
-          const apiKey = "3e1cf70d880acb18f154700af5ac63f8";
-          const url = "https://api.themoviedb.org/3/discover/tv?include_adult=true&include_null_first_air_dates=true&language=en-US&sort_by=popularity.desc" + query + "&page=" + page + "&api_key=" + apiKey;
-          
-          const response = await fetch(url);
-          if (!response.ok) {
-            throw new Error("Network response was not ok");
-          }
-          const data = await response.json();
+      while (!doneDisplaying) {
+        page++;
+        const apiKey = "3e1cf70d880acb18f154700af5ac63f8";
+        const url = `https://api.themoviedb.org/3/discover/tv?include_adult=true&include_null_first_air_dates=true&language=en-US&sort_by=popularity.desc${query}&page=${page}&api_key=${apiKey}`;
+
+        const response = await fetch(url);
+        if (!response.ok) {
+          throw new Error("Network response was not ok");
+        }
+        const data = await response.json();
 
-          let shows = data.results;
+        let shows = data.results;
 
-          shows = await Promise.all(shows.map(async show => {
+        shows = await Promise.all(
+          shows.map(async (show) => {
             const seriesUrl = `https://api.themoviedb.org/3/tv/${show.id}?language=en-US&api_key=${apiKey}`;
             const seriesResponse = await fetch(seriesUrl);
             const seriesData = await seriesResponse.json();
             show.total_episodes = seriesData.number_of_episodes;
             show.rating = seriesData.vote_average;
+
+            // Fetch YouTube trailer info
+            const youtubeUrl = `https://www.googleapis.com/youtube/v3/search?part=snippet&q=${show.name}+trailer&key=${youtubeApiKey}`;
+            const youtubeResponse = await fetch(youtubeUrl);
+            const youtubeData = await youtubeResponse.json();
+            show.youtubeAvailable = youtubeData.items.length > 0;
+            show.youtubeVideoId = youtubeData.items[0]?.id?.videoId;
+
             return show;
-          }));
+          })
+        );
 
-          if(shows.length < 1) { break; } // Stop calling API if there are no more shows
+        // Filter shows with no YouTube trailer or undefined videoId
+        shows = shows.filter((show) => show.youtubeAvailable && show.youtubeVideoId);
```

Comment:
I don't think we should filter out shows if they don't have a trailer. I think instead we should not include a link in the card if that particular show doesn't have a trailer. Please let me know your thoughts.